src/interface/mapping/path.rs

```diff
@@ -0,0 +1,229 @@
+    }
+}
+
+impl<'a> Ord for MappingPath<'a> {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match (self, other) {
+            (MappingPath::Mapping { levels, .. }, MappingPath::Mapping { levels: o_lvs, .. }) => {
+                levels.cmp(o_lvs)
+            }
+            (MappingPath::Endpoint(endpoint), MappingPath::Endpoint(o_endpoint)) => {
+                endpoint.cmp(o_endpoint)
+            }
+            (MappingPath::Mapping { levels, .. }, MappingPath::Endpoint(endpoint)) => {
+                endpoint.cmp_levels(levels).reverse()
+            }
+            (MappingPath::Endpoint(endpoint), MappingPath::Mapping { levels, .. }) => {
+                endpoint.cmp_levels(levels)
+            }
+        }
+    }
+}
+
+impl<'a> From<Endpoint<'a>> for MappingPath<'a> {
+    fn from(endpoint: Endpoint<'a>) -> Self {
+        MappingPath::Endpoint(endpoint)
+    }
+}
+
+impl<'a> Borrow<str> for MappingPath<'a> {
+    fn borrow(&self) -> &str {
+        match self {
+            MappingPath::Mapping { path, .. } => path,
+            MappingPath::Endpoint(endpoint) => endpoint,
+        }
+    }
+}
+
+impl<'a> PartialEq<str> for MappingPath<'a> {
+    fn eq(&self, other: &str) -> bool {
+        match self {
+            MappingPath::Mapping { path, .. } => path == other,
+            MappingPath::Endpoint(endpoint) => endpoint == other,
+        }
+    }
+}
+
+impl PartialEq<&str> for MappingPath<'_> {
+    fn eq(&self, &other: &&str) -> bool {
+        self == other
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, thiserror::Error)]
+pub enum Error {
+    #[error("path missing prefix: {0}")]
+    Prefix(String),
+    #[error("path should have at leas one level")]
+    Empty,
+    #[error("path has an empty level: {0}")]
+    EmptyLevel(String),
+}
+
+impl Error {
+    pub(crate) fn path(&self) -> &str {
+        match self {
+            Error::Prefix(path) => path,
+            Error::Empty => "",
+            Error::EmptyLevel(path) => path,
+        }
+    }
+}
+
+/// Parses the MQTT levels structure of the topic received.
+fn parse_mapping(input: &str) -> Result<MappingPath, Error> {
+    let path = input
+        .strip_prefix('/')
+        .ok_or_else(|| Error::Prefix(input.to_string()))?;
+
+    // Split and check that none are empty
+    let levels: Vec<&str> = path
+        .split('/')
+        .map(|level| {
+            if level.is_empty() {
+                return Err(Error::EmptyLevel(input.to_string()));
+            }
+
+            Ok(level)
+        })
+        .collect::<Result<_, _>>()?;
+
+    if levels.is_empty() {
+        return Err(Error::Empty);
+    }
+
+    Ok(MappingPath::Mapping {
+        path: Cow::Borrowed(input),
+        levels,
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::interface::mapping::endpoint::Level;
+
+    use super::*;
+
+    /// Helper macro to create a `MappingPath` from a string literal.
+    #[macro_export]
+    macro_rules! m {
```

Comment:
Improve the name of this macro even if it is used only in the test environment.

Reply:
renamed in the mapping macro